### DISCUSSION TITLE
useFetch not called on server when deploying to Kubernetes cluster

### DISCUSSION THREAD

**Discussion by JensErikChristiansen** [nuxt/nuxt#17444] [Questions] [answered]
Hi!

We're trying to convert our Vue 3 SPA app into a Nuxt app for SSR.

We have an app where the page is not making Nuxt API calls when being rendered on the server when deploying to a kubernetes cluster. Even when distilling the app down to these files:

/pages/index.vue:
```
<script setup>
const { data: count } = await useFetch('/api/count');
</script>

<template>Page visits: {{ count }}</template>
```

/server/api/count.js:

```
export default defineEventHandler(async function () {
    console.log('Getting count ...');
    return new Promise((resolve, reject) => {
        resolve(10);
    });
});
```

When we view the page, we don't see any results from '/api/count'.

<img width="105" alt="image" src="https://user-images.githubusercontent.com/6833664/173439210-c1a0f719-7408-4403-a207-ad65679f028a.png">

Looking at the logs, we don't see "Getting count …", which indicates that, when the server renders the page, is not even trying to fetch the data.

NOTE that it DOES work when:
	- running a local server ("npm run dev")
	- "npm run build" and "node .output/server/index.mjs"
	- "npm run build" and "npm run preview"
	- Pulling the published docker image that is currently in the kubernetes cluster and running it locally

ANOTHER NOTE: SSR works with useRoute. It appears that the issue lies with API calls or useFetch.

Also note that there is also an app.vue that renders pages/index.vue, and that there is a navbar in layouts/default.vue, but we feel strongly enough that it has no effect on this issue.

Anybody have any ideas as to what we're doing wrong? We can share more info upon request.

Thank you in advance!

**Comment by danielroe** [comment, marked as answer]
Would be good to see more info about that cluster. Can you replicate in a local cluster (minikube or similar) configured similarly?

**Reply by ekawas-td** [reply]
Ok, everything is working as expected. We actually had configured the environment to be behind basic auth and we didn't realize that internal api routes were being blocked too (makes sense now that we know though).
